crates/bevy_ecs/src/schedule_v3/schedule.rs

```diff
@@ -0,0 +1,1027 @@
+use std::{
+    borrow::Cow,
+    fmt::{Debug, Write},
+    result::Result,
+};
+
+#[cfg(feature = "trace")]
+use bevy_utils::tracing::info_span;
+use bevy_utils::{
+    petgraph::{algo::tarjan_scc, prelude::*},
+    thiserror::Error,
+    tracing::{error, info, warn},
+    HashMap, HashSet,
+};
+
+use fixedbitset::FixedBitSet;
+
+use crate::{
+    self as bevy_ecs,
+    component::ComponentId,
+    schedule_v3::*,
+    system::{BoxedSystem, Resource},
+    world::World,
+};
+
+/// Resource that stores [`Schedule`]s mapped to [`ScheduleLabel`]s.
+#[derive(Default, Resource)]
+pub struct Schedules {
+    inner: HashMap<BoxedScheduleLabel, Schedule>,
+}
+
+impl Schedules {
+    /// Constructs an empty `Schedules` with zero initial capacity.
+    pub fn new() -> Self {
+        Self {
+            inner: HashMap::new(),
+        }
+    }
+
+    /// Inserts a labeled schedule into the map.
+    ///
+    /// If the map already had an entry for `label`, `schedule` is inserted,
+    /// and the old schedule is returned. Otherwise, `None` is returned.
+    pub fn insert(&mut self, label: impl ScheduleLabel, schedule: Schedule) -> Option<Schedule> {
+        let label = label.dyn_clone();
+        if self.inner.contains_key(&label) {
+            warn!("schedule with label {:?} already exists", label);
+        }
+        self.inner.insert(label, schedule)
+    }
+
+    /// Removes the `label` entry from the map, returning its schedule if one existed.
+    pub fn remove(&mut self, label: &dyn ScheduleLabel) -> Option<Schedule> {
+        let label = label.dyn_clone();
+        if !self.inner.contains_key(&label) {
+            warn!("schedule with label {:?} not found", label);
+        }
+        self.inner.remove(&label)
+    }
+
+    /// Returns a reference to the schedule associated with `label`, if it exists.
+    pub fn get(&self, label: &dyn ScheduleLabel) -> Option<&Schedule> {
+        let label = label.dyn_clone();
+        self.inner.get(&label)
+    }
+
+    /// Returns a mutable reference to the schedule associated with `label`, if it exists.
+    pub fn get_mut(&mut self, label: &dyn ScheduleLabel) -> Option<&mut Schedule> {
+        let label = label.dyn_clone();
+        self.inner.get_mut(&label)
+    }
+
+    /// Iterates all system change ticks and clamps any older than [`MAX_CHANGE_AGE`](crate::change_detection::MAX_CHANGE_AGE).
+    /// This prevents overflow and thus prevents false positives.
+    ///
+    /// **Note:** Does nothing if the [`World`] counter has not been incremented at least [`CHECK_TICK_THRESHOLD`](crate::change_detection::CHECK_TICK_THRESHOLD)
+    /// times since the previous pass.
+    pub(crate) fn check_change_ticks(&mut self, change_tick: u32) {
```

Comment:
❓ I know what a tick is, but what is a change tick? And why do you need to clamp them?